src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -169,6 +169,45 @@ int otSysMainloopPoll(otSysMainloopContext *aMainloop);
  */
 void otSysMainloopProcess(otInstance *aInstance, const otSysMainloopContext *aMainloop);
 
+#if OPENTHREAD_POSIX_CONFIG_LIB_SHARED_ENABLE
+/**
+ * This function create thread and processing OpenThread stack.
+ *
+ * @param[in,out]   aInstance   A pointer to the OpenThread instance.
+ * @param[in]       aThreadId   A pointer to the thread id.
+ * @param[in]       aComPort    A pointer to the radio device name.
+ * @param[in]       aDebug      A pointer to the debug level.
+ *
+ */
+void otSysGetInstance(otInstance **aInstance, pthread_t *aThreadId, const char *aComPort, uint16_t aDebug);
```

Comment:
Thinking more on this I feel the idea of providing new posix `otSys` APIs which start a `pthread` and/or lock/wait on a mutx may not good to include directly in OT posix platform:
- These are really OS and platform-specific and may be best left for each product/project to decide.
- One of the key goal is that we don't want OT APIs to provide functions of an OS (replace OS).

Reply:
Using openthread shared library, developer can use OT APIs to manage ot instance with multithread support.
Using shared library with multithread support, Iot application can invoke OT APIs directly without handling IPC mechanism in multithread environment. Hence help to improve development time for Iot application.

Note: Commit changes were tested and verified with ioter tool (https://github.com/Samsung/ioter).

The Ioter is a tool that allows you to emulate all Matter supported IoT devices with Linux PC and Thread RCP dongle.